content/blog/how-to-ask-good-questions/index.md

```diff
@@ -0,0 +1,113 @@
+
+## Share what you have tried
+
+Just to save everyone some time, and also get a better understanding of the entire problem, share what you have already tried. If you haven't tried anything yet, try googling for similar problems, and at least attempt to brainstorm a solution. Do your best to try something before asking someone else.
+
+Sharing what you've tried helps people to rule out anything you've already tried and they can concentrate on thinking of another solution that might work. Following my basic example:
+
+> *I've also tried to pass in 2 and 2 and that seemed to work, but then if I passed in 3 and 2 I would get 6.*
+> 
+> 
+> 
+> *Finally I've tried to do sum like the following but that also didn't work:*
+> 
+> ```javascript
+> function sum(a, b) {
+>     return b * a;
+> }
+> ```
+
+
+
+## Other tips for asking questions
+
+Those are the main tips that are applicable to any programming question, regardless of  where / when you're asking the question. But there are other tips that may apply to more specific situations that I want to talk about in this section:
+
+* **Choose the right time and the right person**
+  Are you asking someone at work or in your household? Firstly make sure that it's a good time for them. If they're busy working away then don't disturb them.
+
+* **Do some research**
+  Are there any terms you're unsure of that are relevant to your question? Look around first and make sure *you* know the entire problem before expecting other people to.
+
+* **If something's not clear then ask about it**
+  It's easy to feel bad when someone explains something to you and you still don't fully understand and need to ask again. There is no shame in asking someone to be clearer as long as you're being respectful about it. They're already helping you which means they're probably happy to give away some of their time to help you out!
+
+* **Don't take up too much of someone's time**
+  It's fine if you ask for help and it takes someone 15 minutes to help you out. However if you know this is going to take a longer amount of time, or you notice like the person helping you is starting to get impatient, it's fine to leave it for another time! You could simply ask them when is a more convenient time for you two to have another go at it? Something like:
+  
+  > *"This is taking more time than it thought it would, is it OK if I asked you if we could give it another go maybe when you've got a bit more time?"*
+  
+  Most times people will be happy to!
+
+* **Finally, asking bad questions from time to time is fine**
+  Don't feel bad if you've asked a bad question. It's easy to forget that sometimes Google exists and you ask a coworker something that could've been answered faster by the internet! Similarly, don't lash out when someone asks you a bad question. Instead kindly teach them how to ask good questions for future reference!
```

Comment:
```suggestion
* ###### Choose the right time and the right person
  Are you asking someone at work or in your household? Firstly make sure that it's a good time for them. If they're busy working away then don't disturb them.

* ###### Do some research
  Are there any terms you're unsure of that are relevant to your question? Look around first and make sure *you* know the entire problem before expecting other people to.

* ###### If something's not clear then ask about it
  It's easy to feel bad when someone explains something to you and you still don't fully understand and need to ask again. There is no shame in asking someone to be clearer as long as you're being respectful about it. They're already helping you which means they're probably happy to give away some of their time to help you out!

* ###### Don't take up too much of someone's time
  It's fine if you ask for help and it takes someone 15 minutes to help you out. However if you know this is going to take a longer amount of time, or you notice like the person helping you is starting to get impatient, it's fine to leave it for another time! You could simply ask them when is a more convenient time for you two to have another go at it? Something like:
  
  > *"This is taking more time than it thought it would, is it OK if I asked you if we could give it another go maybe when you've got a bit more time?"*
  
  Most times people will be happy to!

* ###### Finally, asking bad questions from time to time is fine
  Don't feel bad if you've asked a bad question. It's easy to forget that sometimes Google exists and you ask a coworker something that could've been answered faster by the internet! Similarly, don't lash out when someone asks you a bad question. Instead kindly teach them how to ask good questions for future reference!
```

These line breaks inside list items are not preserved in the rendered markdown. In other articles, I've used headings to do this instead - which also changes the margin/font size, making it a bit easier to find things at a glance.